subsys/net/ip/tcp.c

```diff
@@ -2044,6 +2045,14 @@ NET_CONN_CB(tcp_established)
 
 	data_len = net_pkt_appdatalen(pkt);
 	if (data_len > net_tcp_get_recv_wnd(context->tcp)) {
+		/* In case we have zero window, we should still accept
+		 * Zero Window Probes from peer, which per convention
+		 * come with len=1.
+		 */
+		if (data_len == 1 && net_tcp_get_recv_wnd(context->tcp) == 0) {
```

Comment:
`net_tcp_get_recv_wnd(context->tcp) == 0` is redundant here. If  `data_len == 1` and we'e passed `if (data_len > net_tcp_get_recv_wnd(context->tcp))` current receive window size have to be 0.

Reply:
> net_tcp_get_recv_wnd(context->tcp) == 0 is redundant here. If data_len == 1 and we'e passed if (data_len > net_tcp_get_recv_wnd(context->tcp)) current receive window size have to be 0.

That's a little bit too-smartish, I'd say. Will remove the code, but will need to explain it in the comment.